sdk/include/opentelemetry/sdk/common/http_client.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include <chrono>
+#include <string>
+#include <vector>
+#include "opentelemetry/nostd/function_ref.h"
+#include "opentelemetry/version.h"
+
+// Usage Example
+//
+// struct ReponseHandler: public HttpResponseHandler {
+//      void OnHttpResponse(HttpResponse& res) noexcept override
+//      {
+//           if (res.IsSuccess())
+//              .. process response
+//      }
+//  };
+//
+//   HttpSessionManager sessionManager; // implementer can provide singleton implementation for it
+//   auto session = sessionManager.createHttpSession("localhost", 8000);
+//   auto request = session->CreateRequest();
+//   request->AddHeader(..);
+//   ResponseHandler res_handler;
+//   session->SendRequest(res_handler);
```

Comment:
It seems this asynchronously sends a request. How would code look like that synchronously sends a request? How would I go about sending a request and waiting for the response of that request?

Reply:
This is not supported as of now ( was not sure if we need it ). On second thought, there is  nothing wrong in having an option to send synchronous request through the API. Will add it in the api.

Reply:
On ~~second~~ third thought, I felt keeping single method `SendRequest` would make sense, and provides flexibility to implement it as async ( non-blocking ) and sync ( blocking ) mode. In case of synchronous call, the request `session->SendRequest(res_handler)` would return control to next line only after `res_handler` is called.  Let me know your thoughts on that ?